plugins/wallabag.koplugin/main.lua

```diff
@@ -664,25 +664,33 @@ function Wallabag:downloadArticle(article)
     local file_ext = ".epub"
     local item_url = "/api/entries/" .. article.id .. "/export.epub"
 
-    -- The mimetype is actually an HTTP Content-Type, so it can include a semicolon with stuff after it.
-    -- Just in case we also trim it, though that shouldn't be necessary.
-    -- A function represents `null` in our JSON.decode, because `nil` would just disappear.
-    -- We can simplify that to not a string.
-    local mimetype = type(article.mimetype) == string and util.trim(article.mimetype:match("^[^;]*")) or nil
+    -- The mimetype is actually an HTTP Content-Type, so we remove everything from the `;` onward
+    -- Wallabag returns `null` if no mimetype is known, which is JSON.decoded to a function,
+    -- which in turn is converted to `nil` here
+    local mimetype = type(article.mimetype) == "string" and util.trim(article.mimetype:match("^[^;]*")) or nil
 
-    -- If the article links to a supported file type, we will download it directly.
-    -- All webpages are HTML. Ignore them since we want the Wallabag EPUB instead!
     if self.download_original_document then
-        if mimetype ~= "text/html" and DocumentRegistry:hasProvider(nil, mimetype) then
+        -- Download any filetype that we have a provider for, other than HTML, from the original URL
+        if mimetype == "text/html" then
+            logger.dbg("Wallabag:downloadArticle: not ignoring EPUB, because", article.url, "is HTML")
+        elseif mimetype == nil then -- base ourselves on the file extension
+            if util.getFileNameSuffix(article.url):lower() == "html" then
```

Comment:
.htm is or used to be common as well, if this is sufficiently much of a concern.

Reply:
Oh, right, forgot about that one